### DISCUSSION TITLE
Options for private key password encryption

### DISCUSSION THREAD

**Discussion by MrWade0405** [rabbitmq/rabbitmq-server#8109] [Questions] [answered]
Hi All,

I am configuring TLS communication for the RabbitMQ service.
I found that RabbitMQ allows to use password encryption for certificates converted from server certificate and with the specified paths to them in the configuration file (advanced.config).

RabbitMQ says it doesn't make the system more secure, it just satisfies certain country regulations, and I agree with that.

I'm wondering if there is some other way to hide the password that would provide a higher level of configuration security because using even a passphrase is not a good solution for our customers.

Maybe RabbitMQ can read the certificate from certificate store or some other solution?
Thanks in advance.

Best regards,
Vitalii Mahlona

**Comment by michaelklishin** [comment, marked as answer]
Use an encrypted filesystem for your RabbitMQ configuration (see [File and Directory Locations](https://rabbitmq.com/relocate.html)), and/or the `rabbitmq_trust_store` plugin. There are no other options.

**Reply by MrWade0405** [reply]
Thanks for your response.
